test/e2e/common/deploy.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright © 2021 Dell Inc. or its subsidiaries. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+	"path"
+	"time"
+
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2elog "k8s.io/kubernetes/test/e2e/framework/log"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+)
+
+func DeployOperator(c clientset.Interface) (func(), error) {
+	var (
+		executor = CmdHelmExecutor{framework.TestContext.KubeConfig}
+		chart    = HelmChart{
+			name:      "csi-baremetal-operator",
+			path:      path.Join(OperatorTestContext.ChartsFolder, "csi-baremetal-operator"),
+			namespace: "test-csi-operator",
+		}
+		installArgs     = fmt.Sprintf("--set image.tag=%s", OperatorTestContext.OperatorVersion)
+		waitTime        = 1 * time.Minute
+		sleepBeforeWait = 10 * time.Second
+	)
+
+	cleanup := func() {
+		if err := executor.DeleteRelease(&chart); err != nil {
+			e2elog.Logf("CSI Operator helm chart deletion failed. Name: %s, namespace: %s", chart.name, chart.namespace)
+		}
+
+		if err := c.CoreV1().Namespaces().Delete(chart.namespace, nil); err != nil {
+			e2elog.Logf("Namespace deletion failed. Namespace: %s", chart.namespace)
+		}
+	}
+
+	if err := executor.InstallRelease(&chart, installArgs); err != nil {
+		return nil, err
+	}
+
+	time.Sleep(sleepBeforeWait)
```

Comment:
Why do we need to sleep before polling pods?

Reply:
I had lags using `go test` in cli sometimes, when helm has already returned response, but `WaitForPodsRunningReady` has shown 0/0 pods and 0 replicasets in namespace. I have no ideas how to deal with it without Sleep